src/acquisition/covidcast/csv_to_database.py

```diff
@@ -120,6 +120,10 @@ def upload_archive(
         if modified_row_count is None or modified_row_count: # else would indicate zero rows inserted
           total_modified_row_count += (modified_row_count if modified_row_count else 0)
           database.commit()
+      except DBLoadStateException as e:
+        # if the db is in a state that is not fit for loading new data,
+        # then we should stop processing any more files
+        raise e
```

Comment:
Noting for posterity: we don't need a `database.rollback()` here because if this exception is thrown, no changes have yet been made to the database.

Reply:
*yet been made to the db by the file in question, files before this _may_ have been imported correctly

Reply:
wait, really? in theory this exception can only be thrown if acquisition is started when the database is in an inconsistent state. under what scenarios would some files be uploaded successfully before this exception is thrown?

Reply:
its exceedingly unlikely.  but for example, multiple acquisition processes could be running simultaneously

Reply:
...though transaction boundaries would make that virtually impossible